spark/src/main/scala/org/apache/spark/sql/delta/stats/StatsCollectionUtils.scala

```diff
@@ -137,6 +189,21 @@ object StatsCollectionUtils
   }
 }
 
+object ParallelFetchPool {
+  val NUM_THREADS_PER_CORE = 10
+  val MAX_THREADS = 1024
```

Comment:
The thread count is a bit high. Have you seen any issues with your testing?

Reply:
I didn't see any issues when trying it out on a 4 worker cluster. I think you could run into some throttling issues if the cluster you use is very large and you have to collect stats for DVs for many files. But then you can always increase the number of files per partition for stats collection, that should reduce throttling issues.